other/world.ts

```diff
@@ -203,89 +214,95 @@ export default class World {
-      if (distance < bot.radius && bot.radius > otherBot.radius) {
-        botIdsToRemove.push(otherBotSpawnId);
-        // updating statistic
-        const botStats = this.stats.get(bot.botId);
-        if (!botStats) {
-          console.error(new Error("can't find bot stats; check if they were created in `.addBot`"));
+      // check if bot eats other bots
+      const botIdsToRemove: string[] = [];
+      for (const [otherBotSpawnId, otherBot] of this.botSpawns.entries()) {
+        if (otherBotSpawnId === bot.spawnId) {
           continue;
         }
-        const otherBotStats = this.stats.get(otherBot.botId);
-        if (!otherBotStats) {
-          console.error(new Error("can't find other bot stats; check if they were created in `.addBot`"));
-          continue;
+
+        const distance = World.distance(bot, otherBot);
+        if (distance < bot.radius && bot.radius > otherBot.radius) {
+          botIdsToRemove.push(otherBotSpawnId);
+
+          // update stats
+          const botStats = this.stats.get(bot.botId);
+          if (!botStats) {
+            console.error(new Error("can't find bot stats; check if they were created in `.addBot`"));
+            continue;
+          }
+          botStats.kills += 1;
+
+          const otherBotStats = this.stats.get(otherBot.botId);
+          if (!otherBotStats) {
+            console.error(new Error("can't find other bot stats; check if they were created in `.addBot`"));
+            continue;
+          }
+          otherBotStats.deaths += 1;
         }
-        botStats.kills += 1;
-        otherBotStats.deaths += 1;
       }
-    }
 
-    // check if bot eats food
-    const food = this.food;
-    const foodIdxToRemove: number[] = [];
-    for (let i = 0; i < food.length; ++i) {
-      // it's safe to cast to Sprite because know that `i` is within the bounds of the array
-      const distance = World.distance(bot, food[i] as Sprite);
-      if (distance < bot.radius) {
-        foodIdxToRemove.push(i);
-        // updating statistic
-        const botStats = this.stats.get(bot.botId);
-        if (!botStats) {
-          console.error(new Error("can't find bot stats; check if they were created in `.addBot`"));
+      // check if bot eats food
+      const food = this.food;
+      const foodIdxToRemove: number[] = [];
+      for (let i = 0; i < food.length; ++i) {
+        const foodElement = food[i];
+        if (!foodElement) {
+          throw new Error("unexpected: food element is undefined");
+        }
+        const distance = World.distance(bot, foodElement);
+        if (distance < bot.radius) {
+          foodIdxToRemove.push(i);
+
+          // update stats
+          const botStats = this.stats.get(bot.botId);
+          if (!botStats) {
+            console.error(new Error("can't find bot stats; check if they were created in `.addBot`"));
+            continue;
+          }
+          botStats.foodEaten += 1;
+        }
+      }
```

Comment:
So since there have been some changes, i propose to put this logic into separate function - and calling it
As we discussed earlier, it's a good idea to create self-document code through delegation
And yeah, it would be great to do so for other logic blocks

Of course, this isn't a PR just for refactoring but i don't really consider this set of changes as unwanted/unacceptable. If this is inappropriate, let me know

Reply:
@slashnulldot, I agree that this can be separate, but I would keep the logic that decides which food to remove there and only move the stats-related logic away. Keeping all of the collision computation logic in a single function makes it easier to follow, but the stats-updating logic goes in the way.

Reply:
@slashnulldot, I have refactored some of the parts of the `checkCollisions` function in this PR to make it more clear. Thank you for your feedback 🙌